### DISCUSSION TITLE
Rewrite it in Rust?

### DISCUSSION THREAD

**Discussion by Slackadays** [WerWolv/ImHex#914] [General]
With the flurry of activity over at Fish regarding RIIR (https://github.com/fish-shell/fish-shell/pull/9512), I was just wondering if ImHex had any plans to do something similar. 

Considering that Fish had been stuck on C++11 for legacy reasons, I can see where they're coming from because although C++14-23 have done a lot to fix what they're complaining about, they couldn't take advantage of that. Additionally, Rewrite It In Rust™ feels appealing to many people either way. 

On the other hand, RIIR feels like passing gimmick and ImHex is already using the latest C++23 features that are basically as good as Rust. Legacy platform support doesn't matter because those are legacy platforms and are not up to date like they should be.

So, what do you think? I personally have little to no skin in the game because I only care about the end product which is what all the users see.

**Comment by WerWolv** [comment]
Absolutely not lol

While I do like Rust (we even have support for writing your own plugins in Rust), I don't think the massive amount of work would be worth it. Compared to fish, I thankfully have the luxury of just _not caring_ about certain distros. I will keep updating to the latest version of GCC and use the latest C++ features as soon as they become widely available. So if a distro doesn't feel like supporting new GCC versions, I'll simply drop support for it. We're officially only supporting Ubuntu, Fedora, Arch and RedHat anyways which usually get a way to update to the latest compilers pretty quickly. Everybody else can then just use the AppImage.

So long story short, I can see having parts of ImHex being written in Rust as part of a plugin, however I like my C++ too much to switch over :)
